File: android_app_p/app/src/main/java/com/comicviewer/data/model/Manga.kt
package com.comicviewer.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents a manga or comic series from the backend API.
 *
 * Uses Kotlinx Serialization for JSON deserialization with
 * snake_case field mapping.
 */
@Serializable
data class Manga(
    val id: Int,
    val title: String,
    val author: String? = null,
    val description: String? = null,

    @SerialName("cover_path")
    val coverPath: String? = null,

    @SerialName("source_id")
    val sourceId: Int,

    @SerialName("total_chapters")
    val totalChapters: Int = 0,

    @SerialName("created_at")
    val createdAt: String? = null,

    @SerialName("updated_at")
    val updatedAt: String? = null,

    /** Chapters populated from the detail endpoint. */
    val chapters: List<Chapter>? = null
)

File: ios_app_p/ComicViewer/Models/Manga.swift
// Manga.swift
// Comic Viewer – Manga Data Model
//
// Codable model representing a manga / comic series.

import Foundation

/// Represents a manga or comic series from the backend.
struct Manga: Codable, Identifiable, Hashable {
    let id: Int
    let title: String
    let author: String?
    let description: String?
    let coverPath: String?
    let sourceId: Int
    let totalChapters: Int
    let createdAt: String?
    let updatedAt: String?

    /// Chapters associated with this manga (populated from detail endpoint)
    var chapters: [Chapter]?

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case author
        case description
        case coverPath = "cover_path"
        case sourceId = "source_id"
        case totalChapters = "total_chapters"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case chapters
    }

    /// Conformance to Hashable (ignoring chapters for identity)
    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: Manga, rhs: Manga) -> Bool {
        lhs.id == rhs.id
    }
}
